git/odb/blob.go

```diff
@@ -0,0 +1,64 @@
+package odb
+
+import "io"
+
+// Blob represents a Git object of type "blob".
+type Blob struct {
+	// Size is the total uncompressed size of the blob's contents.
+	Size int64
+	// Contents is a reader that yields the uncompressed blob contents. It
+	// may only be read once. It may or may not implement io.ReadSeeker.
+	Contents io.Reader
+
+	// closeFn is a function that is called to free any resources held by
+	// the Blob.  In particular, this will close a file, if the Blob is
+	// being read from a file on disk.
+	closeFn func() error
+}
+
+var _ Object = (*Blob)(nil)
```

Comment:
If that's all, why not just throw it in the comments for each type? These var declarations get flagged one very review that  you add them :)